src/main/scala/no/gdl/bookapi/service/translation/TranslationService.scala

```diff
@@ -9,23 +9,118 @@ package no.gdl.bookapi.service.translation
+    }
+
+    def fetchUpdatesFor(inTranslation: InTranslation): Try[SynchronizeResponse] = {
+      inTranslation.newTranslationId match {
+        case None => Failure(new RuntimeException(s"The book for ${inTranslation.originalTranslationId} has not yet been translated. Cannot fetch updates."))
+        case Some(newTranslationId) => {
+          crowdinClientBuilder.forSourceLanguage(inTranslation.fromLanguage).flatMap(crowdinClient => {
+            readService.withIdAndLanguage(inTranslation.originalTranslationId, inTranslation.fromLanguage) match {
+              case None => Failure(new RuntimeException(s"The original book with id ${inTranslation.originalTranslationId} was not found. Cannot fetch updates."))
+              case Some(originalBook) => {
+                translationRepository.withId(newTranslationId) match {
+                  case None => Failure(new RuntimeException(s"The translated book with id $newTranslationId was not found. Cannot fetch updates."))
+                  case Some(existingTranslation) => {
+                    val inTranslationFilesToUpdate = translationDbService.filesForTranslation(inTranslation.id.get)
+                    val metaDataOpt = inTranslationFilesToUpdate.find(_.fileType == FileType.METADATA).map(metadata => crowdinClient.fetchTranslatedMetaData(metadata, inTranslation.crowdinToLanguage))
+                    metaDataOpt match {
+                      case None => Failure(new RuntimeException(s"No metadata for translation with id ${inTranslation.id} found. Cannot fetch updates"))
+                      case Some(metadataTry) => {
+                        val chapters: Seq[Try[TranslatedChapter]] = inTranslationFilesToUpdate.filter(_.fileType == FileType.CONTENT).map(content => crowdinClient.fetchTranslatedChapter(content, inTranslation.crowdinToLanguage))
+
+                        inTransaction { implicit session =>
+                          val persisted_all = for {
+                            metadata <- metadataTry
+                            persisted <- Try(writeService.updateTranslation(existingTranslation.copy(title = metadata.title, about = metadata.description)))
+                            mergedChapters <- Try(mergeService.mergeChapters(persisted, chapters.filter(_.isSuccess).map(_.get)))
+                            _ <- Try(mergedChapters.map(ch => writeService.updateChapter(ch)))
+                          } yield persisted
+
+                          persisted_all match {
+                            case Success(translation) => Success(SynchronizeResponse(translation.bookId, CrowdinUtils.crowdinUrlToBook(originalBook, crowdinClient.getProjectIdentifier, inTranslation.crowdinToLanguage)))
+                            case Failure(err) => Failure(err)
+                          }
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          })
+        }
+      }
```

Comment:
You can remove the curly braces around the case bodies (4 pairs), for this to look a bit prettier :)